test/common.js

```diff
@@ -0,0 +1,443 @@
+import app           from './server';
+import db            from '../src/models/db';
+import should        from 'should';
+import supertest     from 'supertest';
+
+import * as CONST from './constants';
+import * as ERR from '../src/errors';
+
+const server       = supertest.agent(app);
+const ERRNOS       = ERR.errnos;
+const ERRORS       = ERR.errors;
+
+module.exports = (endpoint, mandatory, optional = [], associations = []) => {
+  const testEntity = CONST[endpoint + 'Entity'];
+  let patchError, patchSuccess, postError, postSuccess;
+
+  const anotherValue = function anotherValue (property) {
+    if (property === 'encodingType') {
+      return testEntity[property];
+    }
+
+    return CONST['another' + property] || testEntity[property] + 'changed';
+  }
+
+  describe('/' + endpoint + ' API', () => {
+    describe('Preconditions', () => {
+      beforeEach(done => {
+        db().then(models => {
+          return Promise.all([
+            models[endpoint].destroy({ where: {} }),
+          ]).then(() => done());
```

Comment:
I guess you need to delete associated models here as well. Or maybe just delete all models.